src/main/java/uk/gov/hmcts/reform/fprl/documentgenerator/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -27,77 +28,86 @@
-        if (!fileName.startsWith(DRAFT_PREFIX)) {
-            fileName = String.join("", DRAFT_PREFIX, fileName);
-        }
-        placeholders.put(IS_DRAFT, true);
+    public GeneratedDocumentInfo generateAndStoreDraftDocument(
+        String templateName,
+        PlaceholderData placeholders,
+        String userAuthToken) {
+        placeholders.getCaseDetails().getCaseData().put(IS_DRAFT, true);
 
-        return getGeneratedDocumentInfo(templateName, placeholders, authorizationToken, fileName);
+        return getGeneratedDocumentInfo(templateName, placeholders, userAuthToken);
     }
 
-    private GeneratedDocumentInfo getGeneratedDocumentInfo(String templateName, Map<String, Object> placeholders,
-                                                           String authorizationToken, String fileName) {
+    private GeneratedDocumentInfo getGeneratedDocumentInfo(
+        String templateName,
+        PlaceholderData placeholders,
+        String userAuthToken) {
         log.debug("Generate and Store Document requested with templateName [{}], placeholders of size [{}]",
-            templateName, placeholders.size());
-        String caseId = getCaseId(placeholders);
+            templateName, placeholders.getCaseDetails().getCaseData().size());
+        String caseId = placeholders.getCaseDetails().getCaseId();
         if (caseId == null) {
             log.warn("caseId is null for template \"" + templateName + "\"");
+        } else {
+            log.info("Generating document for case Id {}", caseId);
         }
 
-        log.info("Generating document for case Id {}", caseId);
-
-        placeholders.put(
+        placeholders.getCaseDetails().getCaseData().put(
             CURRENT_DATE_KEY,
             new SimpleDateFormat(DATE_FORMAT, Locale.getDefault())
-                .format(Date.from(clock.instant())
-                )
+                .format(Date.from(clock.instant()))
         );
 
         byte[] generatedDocument = generateDocument(templateName, placeholders);
         log.info("Document generated for case Id {}", caseId);
-        return storeDocument(generatedDocument, authorizationToken, fileName);
+
+        return storeDocument(generatedDocument, userAuthToken);
     }
 
-    @Override
-    public GeneratedDocumentInfo storeDocument(byte[] document, String authorizationToken, String fileName) {
+    private GeneratedDocumentInfo storeDocument(byte[] document, String userAuthToken) {
         log.debug("Store document requested with document of size [{}]", document.length);
-        return GeneratedDocumentInfoMapper.mapToGeneratedDocumentInfo(
-            evidenceManagementService.storeDocumentAndGetInfo(document, authorizationToken, fileName)
+
+        // we need to map byte[] document to List<MultipartFile> files
+        DocumentUploadRequest documentUploadRequest = new DocumentUploadRequest(
+            "PUBLIC", "C100", "Family Law", new ArrayList<>()
```

Comment:
we need to update it, but I don't know yet how to map `byte[]` to `List<MultipartFile>`